infinite_tracing/lib/infinite_tracing/connection.rb

```diff
@@ -17,6 +17,12 @@
 module NewRelic::Agent
   module InfiniteTracing
     class Connection
+      GZIP_METADATA = {'grpc-internal-encoding-request' => 'gzip',
+                       'grpc-encoding' => 'gzip',
+                       'grpc-accept-encoding' => ['gzip'],
+                       'content-coding' => 'gzip',
+                       'content-encoding' => 'gzip'}.freeze
```

Comment:
To align with [our gist](https://gist.github.com/fallwith/b0971d024f0ed671655cc40967f0c850), use gzip specific metadata.